src/nvc.rs

```diff
@@ -2,6 +2,111 @@ use instruction::*;
 use interconnect::*;
 
 use std::collections::HashSet;
+use std::fmt;
+
+#[derive(Copy, Clone, Default)]
+pub struct CacheEntry {
+    pub tag: u32,
+    pub base_addr: u32,
+    pub subblock_valid: [bool; 2],
+}
+
+impl fmt::Display for CacheEntry {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "base_addr: 0x{:08x}, tag: 0x{:08x}, sb1 valid: {:5}, sb2 valid: {:5}", self.base_addr, self.tag, self.subblock_valid[0], self.subblock_valid[1])
+    }
+}
+
+pub enum CacheResult {
+    Hit,
+    Miss,
+    Disabled,
+}
+
+pub struct Cache {
+    hits: usize,
+    misses: usize,
+    enable: bool,
+    entries: Box<[CacheEntry; 128]>,
+}
+
+impl Default for Cache {
+    fn default() -> Cache {
+        Cache {
+            hits: 0,
+            misses: 0,
+            enable: false,
+            entries: Box::new([CacheEntry::default(); 128]),
+        }
+    }
+}
+
+impl Cache {
+    pub fn clear_entries(&mut self, start: usize, mut count: usize) {
+        if start >= 128 {
+            return;
+        }
+
+        if count > 128 {
+            count = 128;
+        }
+
+        if start + count > 128 {
+            count = 128 - start;
+        }
+
+        for offset in 0..count {
+            self.entries[start + offset].tag = 0;
+            self.entries[start + offset].subblock_valid = [false; 2];
+         }
+    }
+
+    pub fn set_enable(&mut self, enable: bool) {
+        self.enable = enable;
+    }
+
+    pub fn enable(&self) -> bool {
```

Comment:
nit: Since `enable` ends up being a bit ambiguous (sounds a bit like a void fn that will enable the cache rather than a getter) perhaps we should use `is_enabled` for the variable, which would give us `set_is_enabled` and `is_enabled`?